python/pyspark/java_gateway.py

```diff
@@ -147,6 +147,39 @@ def do_server_auth(conn, auth_secret):
         raise Exception("Unexpected reply from iterator server.")
 
 
+def local_connect_and_auth(sock_info):
```

Comment:
@squito, not a big deal but how about `local_connect_and_auth (port, auth_secret)` and ..

```python
(sockfile, sock) = local_connect_and_auth(port, auth_secret)
```

```python
(sock_file, _) = local_connect_and_auth(java_port, auth_secret)
```

```python
port, auth_secret = sock_info
(sockfile, sock) = local_connect_and_auth(port, auth_secret)
```

or 

```python
(sockfile, sock) = local_connect_and_auth(*sock_info)
```

Reply:
done, thanks -- I used the varargs version `*sock_info` for the last one, i forgot about that in python.